### PR TITLE
Delete objects open-api

### PR DESCRIPTION
**Pull request by nopcoder**
Fix https://github.com/treeverse/lakeFS/issues/2768

Adding delete object to lakeFS open-api.
New route under objects that uses POST to send a list of paths to delete (max 1000 items)
Return 204 (no content) in case delete was successful
Return 200 and list of errors - each one with key, status code and message in case delete fails.
Note that object not found is not a failure and will not be reported per object.

### CODE REVIEW

File: api/swagger.yml [hunk 1]
```diff
@@ -198,6 +198,17 @@ components:
...
           example: "main"
 
+    PathList:
+      type: object
+      required:
+        - paths
+      properties:
+        paths:
+          type: array
+          items:
```
Comment (arielshaqed) [inline api/swagger.yml:208]: Not sure what "items" means.

File: api/swagger.yml [hunk 2]
```diff
@@ -478,6 +489,32 @@ components:
...
+      properties:
+        status_code:
+          type: integer
+          description: http status code associated with the path/operation
+        path:
+          type: string
+          description: path to the associated object
+        message:
+          type: string
+          description: short message explaining the error
```
Comment (arielshaqed) [inline api/swagger.yml:506]:
```suggestion
        path:
          type: string
          description: affected path
        status_code:
          type: integer
          description: HTTP status code associated for operation on path
        message:
          type: string
          description: short message explaining status_code
```

Comment (arielshaqed) [reply api/swagger.yml:506]: This puts code and message together, and also changes wording slightly.

File: api/swagger.yml [hunk 3]
```diff
@@ -2592,6 +2629,45 @@ paths:
...
+              $ref: "#/components/schemas/PathList"
+      responses:
+        200:
+          description: Delete objects response
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ObjectErrorList"
+        204:
+          description: object deleted successfully
```
Comment (arielshaqed) [inline api/swagger.yml:2663]:
```suggestion
          description: all requested objects successfully deleted
```
(?)

File: api/swagger.yml [hunk 4]
```diff
@@ -2592,6 +2629,45 @@ paths:
...
+        401:
+          $ref: "#/components/responses/Unauthorized"
```
Comment (arielshaqed) [inline api/swagger.yml:2665]: When do we get these errors?

File: pkg/api/controller.go [hunk 5]
```diff
@@ -76,6 +78,74 @@ type Controller struct {
...
+				Message:    http.StatusText(http.StatusUnauthorized),
+			})
+			continue
+		}
+
+		lg := c.Logger.WithField("path", objectPath)
+		err := c.Catalog.DeleteEntry(ctx, repository, branch, objectPath)
+		switch {
+		case errors.Is(err, catalog.ErrNotFound):
+			lg.Debug("tried to delete a non-existent object")
```
Comment (arielshaqed) [inline pkg/api/controller.go:114]:
```suggestion
			lg.Debug("tried to delete a non-existent object (OK)")
```

**Review by arielshaqed — APPROVED** [review]
Neat; thanks!

File: api/swagger.yml (hunk 1 above)
Comment (nopcoder) [reply api/swagger.yml:208]: Swagger require you to describe each item in the array - the key of the array is 'paths' so each item is a path.
I'll add a description too.

File: api/swagger.yml (hunk 2 above)
Comment (nopcoder) [reply api/swagger.yml:506]: Thanks

File: api/swagger.yml (hunk 4 above)
Comment (nopcoder) [reply api/swagger.yml:2665]: auth middleware

**Pull request merged**